### ISSUE TITLE
Help with development

### ISSUE THREAD

**Issue by j3t** [j3t/jenkins-pipeline-cache-plugin#5]
It would be nice to have some more information about how the plugin can be tested especially during development.

Idea: Having a docker-compose file which starts up Jenkins, MinIO, some plugins and the latest development version of cache plugin.

**Comment by darxriggs** [comment]
Is this still relevant? There are already tests using testcontainers.

If this is about manually trying things, you can create a compose.yml file to start all the required 3rd party services and run `mvn hpi:run` (see http://jenkinsci.github.io/maven-hpi-plugin/run-mojo.html) to start a Jenkins instance with the current plugin development version and its dependencies.

**Comment by j3t** [comment]
If I remember correctly, the hpi:run approach was missing some plugins for proper testing but I haven't checked that in detail yet. I will give it a try, 🙏

**Comment by darxriggs** [comment]
You can add plugins required for manual testing using the `test` scope to the POM file. They will then also be available in the Jenkins instance started by `mvn hpi:run`.

**Comment by j3t** [comment]
I was able to startup Jenkins with all necessary plugins but I have issues with the docker plugin. The agent node is starting up and commands like `sh 'df -h'` are executed and also the result is displayed but as soon as the cache step gets executed, an error is thrown. I don't know exactly why but the tool step, which downloads the JDK, is producing the same kind of error when it gets executed on the remote agent. I also tried JNLP as communication protocol between master and agent, without success.

I will give it one more try but TBH, without a remote agent, it is not worth it I guess (not covered by automated tests yet).
